### ISSUE TITLE
Standardize 'Back' and 'Go back' labels and icons

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#50106]
### Description

There's a bit of inconsistency in the usage of labels for the various 'back' icon buttons used across the interface. Icons are a bit inconsistent as well: sometimes it's the arrow left icon, sometimes it's the chevron left icon.

For better clarity and usability, the labels should be consistent: either always use 'Back' or 'Go back'.

Ideally, the icons should be consistent as well. I'd leave this point to some design feedback though.

A few examples:

In most of the components for mobile and most of the `.native` files, the label is 'Go back`. Not sure how to provide screenshots example for these cases but you can always do a search in the codebase.

Label: 'Go back to the Dashboard' - Chevron icon

<img width="450" alt="Screenshot 2023-04-26 at 16 11 18" src="https://user-images.githubusercontent.com/1682452/234609987-8737b630-e1ff-4e3a-9741-4c7154330aa4.png">

Label: 'Back' - Chevron icon (see also https://github.com/WordPress/gutenberg/pull/50104)

<img width="480" alt="Screenshot 2023-04-26 at 16 03 27" src="https://user-images.githubusercontent.com/1682452/234610596-3ee7c8d6-f6a4-4b08-abba-5889ff6ea93c.png">

Label (visible text): 'Back' - Chevron icon

<img width="805" alt="Back text" src="https://user-images.githubusercontent.com/1682452/234610836-0e6d5a1a-776e-458f-b902-ab48c5cabc30.png">

Label: 'Navigate to the previous view' - Chevron icon

<img width="385" alt="Navigate to the previous view - no tooltip" src="https://user-images.githubusercontent.com/1682452/234611132-583efa7f-2cca-496f-b358-6c325d939afd.png">

Label (visible text): 'Back' - Arrow left icon

<img width="989" alt="post - template editor - text" src="https://user-images.githubusercontent.com/1682452/234611234-68ee8eb7-df85-422c-8eb3-6f731f453423.png">

Label (visible text): 'Back' - Arrow left icon

<img width="1011" alt="editing template part from template" src="https://user-images.githubusercontent.com/1682452/234611377-35d64803-0e44-4730-875d-75fca7232f31.png">

### Step-by-step reproduction instructions

- Search the codebase for translatable strings that start with either 'back' or 'go back'.
- Navigate the Post Editor and Site Editor interface and observe the various 'back' buttons to have inconsistent labels and icons.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by xerpa43** [comment]
Hi, I would love to work on this one. I am new and this seems like something which I could do

**Comment by afercia** [comment]
@xerpa43 Hi and welcome. Do feel free to go ahead and submit a pull request! 🎉 For some initiali guidance: https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/README.md

**Comment by jasmussen** [comment]
Good one. I'd personally lean towards standardizing on the chevron and the "Back" tooltip across as many instances as possible.

The "← Back" in focused editing could probably use a bit of a re-think. It seems like it is duplicating the W button in its behavior now, and might even be possible to retire. It does also, as far as I recall, serve as a "back to the page editor" when you enter this section from editing a page, so this is worth looking at with care.

The "Back to the dashboard" tooltip, is it sufficient to replace this with just "Back"?

**Comment by autumnfjeld** [comment]
Does anyone know why an icon has been used in the top left to represent "Back to the Dashboard"?   This is no intuitive and from my point of view is only learned thru "accidental button clicking".  Why isn't a clear Back icon or Back text used here?

In this example the green thing is my site icon, serving as "Back"
![image](https://github.com/WordPress/gutenberg/assets/5665959/ff782df8-0d3c-4772-939e-18598bc96eb2)

In this example the WordPress logo is the "Back"
<img width="301" alt="image" src="https://github.com/WordPress/gutenberg/assets/5665959/bf529160-5403-473f-9309-a80a8a9079a0">

-----
Also this is odd behavior, a bug?  I'm clicking on my green-flower icon repeatedly and caught in a loop.
https://d.pr/v/i9q4IC


https://github.com/WordPress/gutenberg/assets/5665959/f742306f-ced0-47f3-b2c1-954d2b001ac4

-------
Repeating one of the above comments, in the Site Editor browser view, both the W icon and the `<`  go back to the Dashboard:

<img width="358" alt="image" src="https://github.com/WordPress/gutenberg/assets/5665959/e51ef4d3-1f8e-46f1-ba1c-7b6719055136">

<img width="363" alt="image" src="https://github.com/WordPress/gutenberg/assets/5665959/bd92ba45-dac0-4ae4-89f7-b128f64c0d43">



-------

**Comment by afercia** [comment]
@autumnfjeld good point. Personally, I find the W logo / Site icon button very confusing and inconsistent. Depending on the editor and its state, it is used for different actions which is less than ideal. Basically, you click on it and you never know what is going to happen. Does it open the navigation panel? Does it go back to the Dashboard? Does it go back to the Posts list?
Also, the default W logo isn't the best image to communicate the udnerlying functionality. Historically, in WordPress core the W logo has always been used for other purposes and users are used to that. In my opinion, it is even worse when the W logo gets replaced by the Site icon. To me, a button or link with the Site icon just means thtat clicking on it I g'm brought to the front end site.
That said, I'd think all these points are slightly off-toopic as this issue only aims to make the Back label and icon consistent across the editor. Please do feel free to create a separate issue, if you like, and report your thoughts / struggles witht the W logo / Site icon.

**Comment by richtabor** [comment]
Related to https://github.com/WordPress/gutenberg/issues/54628.
